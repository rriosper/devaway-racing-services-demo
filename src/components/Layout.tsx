import styled, { css } from '@xstyled/styled-components';
import { up } from '@xstyled/system';
import React from 'react';

const SLayout = styled.div`
  padding-left: lg;
  padding-right: lg;
  padding-bottom: md;
  width: 100%;
  height: 100%;

  ${up(
    'xl',
    css`
      padding-left: 0;
      padding-right: 0;
      max-width: 1180px;
      margin: 0 auto;
    `,
  )}
`;

type LayoutProps = {
  title: React.ReactNode;
  subTitle: React.ReactNode;
  children: React.ReactNode;
};

const Layout: React.FC<LayoutProps> = ({ title, subTitle, children }) => (
  <SLayout>
    {title}
    {subTitle}
    {children}
  </SLayout>
);

export default Layout;
