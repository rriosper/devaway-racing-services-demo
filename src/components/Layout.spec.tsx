import React from 'react';
import { render } from '@testing-library/react';

import Layout from './Layout';

test('Layout', () => {
  const { asFragment, getByText } = render(
    <Layout title={<div>Title</div>} subTitle={<div>Subtitle</div>}>
      CHILDREN
    </Layout>,
  );
  expect(getByText('Title')).toBeInTheDocument();
  expect(getByText('Subtitle')).toBeInTheDocument();
  expect(getByText('CHILDREN')).toBeInTheDocument();

  expect(asFragment).toMatchSnapshot();
});
