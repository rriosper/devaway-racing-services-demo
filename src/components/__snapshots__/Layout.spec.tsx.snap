// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout 1`] = `[Function]`;
